src/Upgrade/Infrastructure/VersionControlSystem/Generator/PullRequestDataGenerator.php

```diff
@@ -45,116 +60,256 @@ public function buildBody(
      *
      * @return string
      */
-    protected function createDescriptionAboutText(?int $releaseGroupId, ?string $jiraIssueLink = null): string
+    protected function buildReleaseGroupsTable(
+        ConfigurationProviderInterface $upgradeConfigurationProvider,
+        StepsResponseDto $stepsResponseDto,
+        ?int $releaseGroupId
+    ): string {
+        if (count($stepsResponseDto->getAppliedReleaseGroups()) === 0) {
+            return '';
+        }
+
+        $isSequentialProcessor = $upgradeConfigurationProvider->getReleaseGroupProcessor(
+        ) === ConfigurationProviderInterface::SEQUENTIAL_RELEASE_GROUP_PROCESSOR;
+
+        $shouldDisplayWarningColumn = $releaseGroupId !== null || $isSequentialProcessor;
+        $shouldDisplayRatingColumn = $this->integratorEvaluator->isIntegratorShouldBeInvoked();
+
+        $text = sprintf(
+            '| Release |%s%s',
+            $shouldDisplayRatingColumn ? ' Efforts saved by Upgrader |' : '',
+            $shouldDisplayWarningColumn ? ' Warnings detected? |' : '',
+        ) . PHP_EOL;
+
+        $text .= sprintf(
+            '| ------- |%s%s',
+            $shouldDisplayRatingColumn ? ' ---- |' : '',
+            $shouldDisplayWarningColumn ? ' ------------------ |' : '',
+        ) . PHP_EOL;
+
+        foreach ($stepsResponseDto->getAppliedReleaseGroups() as $appliedReleaseGroup) {
+            $text .= sprintf(
+                '| [%s](%s) |%s%s',
+                $appliedReleaseGroup->getId(),
+                $appliedReleaseGroup->getLink(),
+                $shouldDisplayRatingColumn ? $appliedReleaseGroup->getRating() . ' |' : '',
+                $shouldDisplayWarningColumn ? $this->getReleaseGroupsTableWarningColumnText(
+                    $stepsResponseDto,
+                    $appliedReleaseGroup,
+                ) . ' |' : '',
+            ) . PHP_EOL;
+        }
+
+        $text .= PHP_EOL;
+
+        return $text;
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsResponseDto
+     * @param \ReleaseApp\Infrastructure\Shared\Dto\ReleaseGroupDto $appliedReleaseGroup
+     *
+     * @return string
+     */
+    protected function getReleaseGroupsTableWarningColumnText(
+        StepsResponseDto $stepsResponseDto,
+        ReleaseGroupDto $appliedReleaseGroup
+    ): string {
+        return $this->responseHasWarnings($stepsResponseDto, $appliedReleaseGroup) ? 'Yes :warning:' : 'No';
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsResponseDto
+     * @param \ReleaseApp\Infrastructure\Shared\Dto\ReleaseGroupDto $releaseGroupDto
+     *
+     * @return bool
+     */
+    protected function responseHasWarnings(StepsResponseDto $stepsResponseDto, ReleaseGroupDto $releaseGroupDto): bool
     {
-        $text = 'Auto created via Upgrader tool';
+        $releaseGroupId = $releaseGroupDto->getId();
+
+        return (
+                isset($stepsResponseDto->getBlockers()[$releaseGroupId])
```

Comment:
please implement an additional method to the `StepResponseDto` so it will be
```suggestion
                $blockers = $stepsResponseDto->getBlockersByIReleaseGroupId($releaseGroupId);
                ...
```
The same for the violations and Integrator response options

Reply:
done